src/main/java/core/org/akaza/openclinica/service/crfdata/xform/PdfActionUrlObject.java

```diff
@@ -61,7 +61,7 @@ public PdfActionUrlObject(FormLayout formLayout, String crfOid, String instance,
                               EventCrf eventCrf, String loadWarning, boolean formLocked, String studySubjectOID, String format,
                               String margin, String landscape) {
         super(formLayout, crfOid, instance, ecid, redirect, markComplete, studyOid, mediaList, goTo, flavor, role,
```

Comment:
Instead of passing null for the interface, can you have two constructors in the super class such that you can keep the old constructor as is and add a new one that take interface as an additional argument? That way, we don't need to change the classes that don't need this interface option and avoid passing nulls.